x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/dataframe/inference/InferenceRunnerTests.java

```diff
@@ -185,7 +190,93 @@ private InferenceRunner createInferenceRunner(ExtractedFields extractedFields) {
             config,
             extractedFields,
             progressTracker,
-            new DataCountsTracker(new DataCounts(config.getId()))
+            new DataCountsTracker(new DataCounts(config.getId())),
+            id -> testDocsIterator
+        );
+    }
+
+    private static Client mockClient() {
+        var client = mock(Client.class);
+        var threadpool = mock(ThreadPool.class);
+        when(threadpool.getThreadContext()).thenReturn(new ThreadContext(Settings.EMPTY));
+        when(client.threadPool()).thenReturn(threadpool);
+        var searchResponse = new AtomicReference<Supplier<SearchResponse>>(
```

Comment:
Hmm I wonder if we can use a plain action future here 🤔 

Would something like this work? https://github.com/elastic/elasticsearch/blob/7fc1fcb2eedcf16e63247157a81895ef812be7e5/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ClientHelperTests.java#L249-L253

Reply:
We could, though we still need to jump through a few hopes for the SearchResponse's leak detector.